corehq/apps/accounting/tests/test_invoice_factory.py

```diff
@@ -136,9 +137,10 @@ def test_paused_plan_generates_no_invoice(self):
             self.account, self.domain.name, paused_plan,
             date_start=self.invoice_start,
             date_end=self.invoice_end + datetime.timedelta(days=1),
-
         )
-        self.assertListEqual(self.invoice_factory._get_subscriptions(), [])
+        with patch.object(self.invoice_factory, '_create_invoice_for_subscription') as mock_create_invoice:
```

Comment:
if `_create_invoice_for_subscription` was renamed, this test might still pass (but incorrectly)? Good to make sure that this fails if `_create_invoice_for_subscription` is renamed and called. Or consider adding a test case / rewriting to test against invoice objects directly.

Reply:
can also check how `patch` behaves if the method isn't there at all...

Reply:
`patch.object` does indeed fail if the method it's patching doesn't exist:
```
AttributeError("<corehq.apps.accounting.invoicing.DomainInvoiceFactory object at 0x165e88070> does not have the attribute 'attr_that_doesnt_exist'")
```

Reply:
Sitting with this for a bit, I think checking that `_create_invoice_for_subscription` was not called is actually more explicit here than checking that no invoices were created, because we don't necessarily know what led to no invoices being created. My preference is to leave it as is. Could also see an argument for doing something like testing that `mock_create_invoice` wasn't called _with the subscription in question_; that would look something like:
```python
        subscription = Subscription.new_domain_subscription(...)
        with patch.object(self.invoice_factory, '_create_invoice_for_subscription') as mock_create_invoice:
            self.invoice_factory.create_invoices()
            self.assertNotIn(mock.call(subscription), mock_create_invoice.call_args_list)
```
Thoughts? @mjriley @jingcheng16

Reply:
@nospame so, our thought is that since `test_paused_plan_generates_no_invoice` specifically points out that we want to make sure no invoices were generated...we should test exactly that, as the logic surrounding the invoice generation could also change. Is it too expensive to generate invoices and check on them here?

Reply:
To me, it needs to be contrasted with something that _does_ generate an invoice, so I did that next to a Community plan that would generate one. How's this? https://github.com/dimagi/commcare-hq/pull/35354/commits/33ffa04b08687b5a5b6b2114bf3061367b1edf0f
@mjriley @jingcheng16

Reply:
looks good!